packages/core/tests/mysqlQueue.test.ts

```diff
@@ -125,58 +126,139 @@ describe("mysqlQueue", () => {
+          failedAt: null,
+          id: expect.any(String),
+          latestFailureReason: null,
+          name: "test_job",
+          payload: { message: "Hello, world!" },
+          priority: 0,
+          queueId: expect.any(String),
+          startAfter: null,
+          status: "pending",
+        });
       });
 
-      const [row] = await queryDatabase.query<RowDataPacket[]>(`SELECT * FROM ${instance.jobsTable()};`);
+      it("should throw case queue not exists", async () => {
+        const unknownQueueName = "anotherQueue";
 
-      expect(isValidUUID(jobIds[0])).toBeTruthy();
-      expect(row).toEqual({
-        attempts: 0,
-        completedAt: null,
-        createdAt: expect.any(Date),
-        failedAt: null,
-        id: expect.any(String),
-        latestFailureReason: null,
-        name: "test_job",
-        payload: { message: "Hello, world!" },
-        priority: 0,
-        queueId: expect.any(String),
-        startAfter: null,
-        status: "pending",
+        await expect(
+          instance.enqueue(unknownQueueName, {
+            name: "test_job",
+            payload: { message: "Hello, world!" },
+          }),
+        ).rejects.toThrowError("Failed to add jobs, maybe queue does not exist");
       });
-    });
 
-    it("should throw case queue not exists", async () => {
-      const unknownQueueName = "anotherQueue";
+      it("should fire the worker callback", async () => {
+        const promise = instance.getJobExecutionPromise(queueName);
 
-      await expect(
-        instance.enqueue(unknownQueueName, {
+        const workerCbMock = vi.fn();
+        const worker = await instance.work(queueName, workerCbMock);
+
+        const { jobIds } = await instance.enqueue(queueName, {
           name: "test_job",
           payload: { message: "Hello, world!" },
-        }),
-      ).rejects.toThrowError("Failed to add jobs, maybe queue does not exist");
-    });
+        });
 
-    it("should fire the worker callback", async () => {
-      const promise = instance.getJobExecutionPromise(queueName);
+        void worker.start();
+        await promise;
 
-      const workerCbMock = vi.fn();
-      const worker = await instance.work(queueName, workerCbMock);
+        await worker.stop();
+        expect(workerCbMock).toHaveBeenCalledWith(expect.objectContaining({ id: jobIds[0] }), expect.anything(), expect.anything());
+      });
+    });
 
-      const { jobIds } = await instance.enqueue(queueName, {
-        name: "test_job",
-        payload: { message: "Hello, world!" },
+    describe("with external db", () => {
+      const externalDb: ExternalDb = {
+        async query(sql, paramteres) {
+          const connection = await queryDatabase.pool.getConnection();
+          const result = await connection.query(sql, paramteres);
+          connection.release();
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          return { affectedRows: (result[0] as any).affectedRows };
```

Comment:
```suggestion
          return { affectedRows: (result[0] as { affectedRows: number }).affectedRows };
```
just for clarity and type safety